### PR TITLE
add url extractor

### PR DESCRIPTION
**Pull request by ZwEin27**

### CODE REVIEW

File: requirments.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Babel==2.3.4
```
Comment (jasonslepicka) [inline requirments.txt]: This file should be *requirements.txt* not *requirments.txt*

File: requirments.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+decorator==4.0.10
+digExtractor>=0.1.8
+esmre==0.3.1
+idna==2.1
+jsonpath-rw==1.4.0
```
Comment (jasonslepicka) [inline requirments.txt]: Can you go through and make sure these packages are explicitly required by this extractor and remove them if they aren't?  Also remove any transitive dependencies like the jsonpath stuff.

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+from setuptools import Extension,find_packages
+from os import path
+
+setup(
+    name = 'digURLExtractor',
+    version = '0.1.0',
+    description = 'digURLExtractor',
+    author = 'Lingzhe Teng',
+    author_email = 'zwein27@gmail.com',
+    url = 'https://github.com/ZwEin27/dig-url-extractor',
```
Comment (jasonslepicka) [inline setup.py]: These should be the usc-isi-i2 urls

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    download_url = 'https://github.com/ZwEin27/dig-url-extractor',
+    packages = find_packages(),
+    keywords = ['url', 'extractor'],
+    install_requires=['digSparkUtil', 'digExtractor', 'esmre', 'idna', 'tldextract']
```
Comment (jasonslepicka) [inline setup.py]: Can you remove digSparkUtil and add any other packages you explicitly need to run the extractor if necessary?

File: digURLExtractor/ze_url_extractor.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        return ans
+
+    @staticmethod
+    def extract(text):
+        text = text.encode('ascii', 'ignore')
+        text= ZEURLExtractor.preprocess(text)
+        ans = ZEURLExtractor.query(text)
+        return ans
+
+if __name__ == '__main__':
```
Comment (jasonslepicka) [inline digURLExtractor/ze_url_extractor.py]: This should be removed.

File: digURLExtractor/url_extractor.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        return copy.copy(self.metadata)
+
+    def set_metadata(self, metadata):
+        self.metadata = metadata
+        return self
+
+    def get_renamed_input_fields(self):
+        return self.renamed_input_fields
+
+    def set_renamed_input_fields(self, renamed_input_fields):
```
Comment (jasonslepicka) [inline digURLExtractor/url_extractor.py]: This can be removed since it's inherited

**Comment by ZwEin27** [comment]
fixed issues for url extractor, thank you,:)

**Pull request merged**